### PR TITLE
[macOS] Group per-view information in `FlutterCompositor` into a class

### PR DESCRIPTION
**Pull request by dkwingsmt**
This PR groups per-view information in `FlutterCompositor` into a private class, `ViewPresenter`. This makes it easier to manage per-view data and write view operations.

Part of https://github.com/flutter/flutter/issues/145874.

Currently, view presenters are never removed once created, since the macOS runner doesn't support removing views. This will be added in the future.

## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I added new tests to check the change I am making or feature I am adding, or the PR is [test-exempt]. See [testing the engine] for instructions on writing and running engine tests.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I signed the [CLA].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by dkwingsmt** [comment]
@knopp Can you take a look? In addition to the change itself, how do you think I can test it (and further changes to the composition code)? Because macOS `FlutterCompositor` currently doesn't have any test harness.

File: shell/platform/darwin/macos/framework/Source/FlutterCompositor.h [hunk 1]
```diff
@@ -56,19 +57,32 @@ class FlutterCompositor {
...
-                            const std::vector<PlatformViewLayerWithIndex>& platform_views_layers);
-
-  // Presents the platform view layer represented by `layer`. `layer_index` is
-  // used to position the layer in the z-axis. If the layer does not have a
-  // superview, it will become subview of `default_base_view`.
-  FlutterMutatorView* PresentPlatformView(FlutterView* default_base_view,
-                                          const PlatformViewLayer& layer,
-                                          size_t index);
+  // A class that contains the information for a view to be presented.
+  class ViewPresenter {
```
Comment (knopp) [inline shell/platform/darwin/macos/framework/Source/FlutterCompositor.h:65]: This should probably be FML_DISALLOW_COPY_AND_ASSIGN?

File: shell/platform/darwin/macos/framework/Source/FlutterCompositor.mm [hunk 2]
```diff
@@ -98,24 +101,32 @@
...
+    const std::vector<PlatformViewLayerWithIndex>& platform_views,
+    const FlutterPlatformViewController* platform_view_controller) {
   FML_DCHECK([[NSThread currentThread] isMainThread])
       << "Must be on the main thread to present platform views";
 
   // Active mutator views for this frame.
   NSMutableArray<FlutterMutatorView*>* present_mutators = [NSMutableArray array];
 
   for (const auto& platform_view : platform_views) {
-    [present_mutators addObject:PresentPlatformView(default_base_view, platform_view.first,
```
Comment (knopp) [inline shell/platform/darwin/macos/framework/Source/FlutterCompositor.mm:117]: Just a nit - The formatting looks bit weird after adding additional argument. Maybe use a temporary variable for the result of PresentPlatformView?

**Comment by knopp** [comment]
Looks good, left couple of nits, is this still a draft?

**Comment by dkwingsmt** [comment]
@knopp I'm mostly wondering if there's any unit tests you'd like me to add. I'm also ok with skipping this one and testing the macOS engine as a whole in a later patch.

**Comment by knopp** [comment]
Right. It is currently only tested in `TEST_F(FlutterEngineTest, Compositor)`. There were tests before but they were kinda useless so @cbracken has removed them. https://github.com/flutter/flutter/issues/137648

I think this is a fairly minor refactor I'd be okay with this landing without tests.

**Review by knopp — APPROVED** [review]

**Pull request merged**